### PR TITLE
PIC-4279: Changes to cater for when hearings can have multiple cases

### PR DESCRIPTION
**Pull request by jpanesar07**
PIC-4279: Changes to cater for when hearings can have multiple cases associated with the same hearing id

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcaseservice/controller/CourtCaseController.java [hunk 1]
```diff
@@ -277,6 +277,16 @@ ExtendedHearingRequestResponse getHearingByHearingId(@PathVariable(value = "hear
         return ExtendedHearingRequestResponse.of(hearingEntity);
     }
 
+    @Operation(description = "Returns extended court case data, by hearing id.")
```
Comment (Obsiye) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/controller/CourtCaseController.java:280]: minor change to include `and court id.`

Comment (jpanesar07) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/controller/CourtCaseController.java]: Good sport, added

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/HearingEntityInitService.java [hunk 2]
```diff
@@ -72,8 +84,9 @@ public Optional<HearingEntity> findFirstByHearingIdFileUpload(String hearingId)
     }
 
     @Transactional
-    public Optional<HearingEntity> findHearingByHearingIdAndDefendantIdInitialiseCaseDefendants(String hearingId, String defendantId) {
-        var hearing = hearingRepository.findFirstByHearingId(hearingId);
+    public Optional<HearingEntity> findAllHearingByHearingIdAndDefendantIdInitialiseCaseDefendants(String hearingId, String defendantId) {
```
Comment (Obsiye) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/HearingEntityInitService.java:87]: hi, it says findAll but the function returns 1 HearingEntity. Should it be `findHearingByHearingIdAndDefendantIdInitialiseCaseDefendants` ?

File: src/test/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepositoryFacadeIntTest.java [hunk 3]
```diff
@@ -58,6 +58,20 @@ public void whenFindFirstByHearingIdOrderByIdDesc_thenReturnCorrectRecordWithOff
...
 
+    @Test
+    public void whenFindFirstByHearingIdAndCourtCaseId_thenReturnCorrectRecordWithOffender() {
+        final var actual = hearingRepositoryFacade.findFirstByHearingIdAndCourtCaseId("5564cbfd-3d53-4f36-9508-437416b08738", "727af2a3-f9ec-4544-b5ef-2ec3ec0fcf2b");
+
+        assertIsFerrisBueller(actual);
+    }
+
+    @Test
+    public void whenFindFirstByHearingIdAndCourtCaseId_thenDoNotReturnRecord() {
```
Comment (Obsiye) [inline src/test/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepositoryFacadeIntTest.java:69]: I'm assuming `"827af2a3-f9ec-4544-b5ef-2ec3ec0fcf2b"` does not exist. 

is there an Integration test that has a hearing with a court case in the database and then we do a PUT /hearing with multiple cases (1 of them being the one that exists and the other doesn't)

testing the PUT

**Review by Obsiye — APPROVED** [review]

**Pull request merged**